### DISCUSSION TITLE
Lesson 3: Does unused imports actually mean higher gas usage?

### DISCUSSION THREAD

**Discussion by spencerj411** [Cyfrin/foundry-full-course-cu#1360] [Q&A]
tl;dr Contrary to what the video says, unused imports don't use more gas. Please let me know your thoughts/if I've misunderstood.

From around https://youtu.be/umepbfKp5rI?t=13467, it was explained that one reason why you should use named imports is because it adds computational complexity to your smart contract, and therefore, will result in higher gas fees. However, from my understanding, with the help of ChatGPT, whilst of course it is bad coding practice to import code you don't intend to use, solc (solidity's compiler) is smart enough to ignore imports that are not used (it uses a process called 'dead code elimination'). Therefore, the unused imports don't/shouldn't actually add to the complexity of your contract during deployment on the blockchain.

I'd love to hear other people's thoughts on this/to be corrected. Thanks in advance!
